engine/glfw/lib/cocoa/cocoa_window.m

```diff
@@ -150,9 +175,21 @@ - (void)activateProcess
 - (void)activateWindow
 {
     [_glfwWin.window makeKeyAndOrderFront:nil];
+    // Starting from macOS Sonoma (14.0) an extra step needed
+    NSOperatingSystemVersion version = [[NSProcessInfo processInfo] operatingSystemVersion];
+    if (version.majorVersion >= 14) {
+        [NSApp activateIgnoringOtherApps:YES];
+    }
     [[NSRunningApplication currentApplication] activateWithOptions:(NSApplicationActivateAllWindows | NSApplicationActivateIgnoringOtherApps)];
 }
 
+// macos 12+
+// https://sector7.computest.nl/post/2022-08-process-injection-breaking-all-macos-security-layers-with-a-single-vulnerability/
+- (BOOL)applicationSupportsSecureRestorableState:(NSApplication *)app
+{
+    return YES;
+}
```

Comment:
another warning in terminal, it's not directly related but it's nice to fix it all together

Reply:
It mentions macos 12. We support BigSur which is 11, will it work?

Reply:
I think so, because it's just a function in delegate. I can put any function with any name there, so on macos < 12 it should be just a never called function